gnucash/import-export/aqb/gnc-gwen-gui.c

```diff
@@ -1425,28 +1466,43 @@ getpassword_cb(GWEN_GUI *gwen_gui, guint32 flags, const gchar *token,
     if(is_tan && methodId == GWEN_Gui_PasswordMethod_OpticalHHD)
     {
         /**
-        * TODO: How to handle Flicker code (use WebView and JS???)
-        *
         * use GWEN_Gui_PasswordMethod_Mask to get the basic method id
         *  cf. gui/gui.h of gwenhywfar
         */
         opticalMethodId=GWEN_DB_GetIntValue(methodParams, "tanMethodId", 0, AB_BANKING_TANMETHOD_TEXT);
         switch(opticalMethodId)
         {
+            case AB_BANKING_TANMETHOD_CHIPTAN:
+                break;
+            case AB_BANKING_TANMETHOD_CHIPTAN_OPTIC:
+                mimeType = "text/x-flickercode";
+                pChallenge = GWEN_DB_GetCharValue(methodParams, "challenge", 0, NULL);
+                if ((NULL == pChallenge) || ('\0' == pChallenge [0]))
```

Comment:
Don't put the constant first in conditionals. That almost made sense in 1975 but compilers are smart enough now to catch when you forget the second `=` so there's no good reason to make the code harder to read.